src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  const stateClone = { ...state };
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(stateClone, key.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const removeKey of key.keysToRemove) {
+          delete stateClone[removeKey];
+        }
+        break;
+
+      case 'clear':
+        for (const kay in stateClone) {
+          delete stateClone[kay];
```

Comment:
[CODE STYLE] - When clearing the state, instead of deleting each property individually inside a loop, you can simply assign an empty object to 'stateClone'.